tests/acceptance/test_dbus.py

```diff
@@ -24,9 +24,75 @@
 class TestDBUS:
     @pytest.mark.min_mender_version("2.6.0")
     def test_dbus_system_configuration_file(self, bitbake_variables, connection):
-        """Test that dbus configuration file is correctly installed."""
+        """Test that DBus configuration file is correctly installed."""
 
         output = connection.run(
             "cat /usr/share/dbus-1/system.d/io.mender.AuthenticationManager.conf"
         ).stdout.strip()
         assert "io.mender.AuthenticationManager" in output
+
+    @pytest.mark.min_mender_version("2.6.0")
+    def test_dbus_get_jwt_token(self, bitbake_variables, connection):
+        """Test JWT token can be retrieved using DBus."""
+        conffile = "/data/etc/mender/mender.conf"
+        backup = "%s.backup" % conffile
+        bdir = os.path.dirname(backup)
+        connection.run(
+            "mkdir -p %s && if [ -e %s ]; then cp %s %s; fi"
+            % (bdir, conffile, conffile, backup)
+        )
+
+        with open("tmpfile", "w") as fd:
+            fd.write(
+                """{
+    "ClientProtocol": "https",
+    "HttpsClient": {
+        "SkipVerify": true
+    },
+    "InventoryPollIntervalSeconds": 5,
+    "RetryPollIntervalSeconds": 5,
+    "ServerCertificate": "/usr/share/doc/mender-client/examples/demo.crt",
+    "SkipVerify": true,
+    "ServerURL": "https://localhost:8443",
+    "DBus": {
+        "Enabled": true
+    }
+}
+"""
+            )
+        try:
+            put_no_sftp("tmpfile", connection, remote=conffile)
+        finally:
+            os.remove("tmpfile")
+
+        result = connection.run("mender bootstrap --forcebootstrap")
+        assert result.exited == 0
+
+        result = connection.run("systemctl start mender-client")
+        assert result.exited == 0
+
+        # this is a portion of the JWT token served by the Mender mock server:
+        # see: meta-mender-ci/recipes-testing/mender-mock-server/mender-mock-server.py
+        token = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9."
+
+        try:
```

Comment:
Suggest moving `try` up to right before the `mender bootstrap` command. That guarantees that everything will be cleaned up even if something fails there, and `systemctl stop` is safe to call on an already stopped service.

Reply:
Done